waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -534,9 +534,9 @@
   (let [home-path (str "/home/" run-as-user)
         base-env (scheduler/environment service-id service-description
                                         service-id->password-fn home-path)
-        ;; Randomize $PORT0 value to ensure clients can't hardcode it.
+        ;; Make $PORT0 value pseudo-random to ensure clients can't hardcode it.
         ;; Helps maintain compatibility with Marathon, where port assignment is dynamic.
-        port0 (-> (rand-int 100) (* 10) (+ pod-base-port))
+        port0 (-> service-id hash (mod 100) (* 10) (+ pod-base-port))
```

Comment:
Nitpick: Why do we want the port number to end in zero? wouldn't doing `(-> service-id hash (mod 1000) (+ pod-base-port))` increase the port range?

Reply:
I found it odd since don't have that restriction on Marathon services, but am okay with whatever we decide.